### PR TITLE
Allow users to add pics through camera or from image gallery

### PR DESCRIPTION
**Pull request by HellinSun**
- Implemented a dialog to allow users to choose between camera and gallery
- if camera, store the picture taken locally and display it in the recycle view.
- if gallery, allow the users to select multiple pictures at once.

### CODE REVIEW

**Review by arfi2212 — APPROVED** [review]
Great job Hellin! :)

File: app/src/main/java/com/example/ar_reshare/AddProduct.java [hunk 1]
```diff
@@ -1,51 +1,109 @@
...
+
+public class AddProduct extends AppCompatActivity implements addPhotoDialog.NoticeDialogListener{
+
+    private ArrayList<Uri> uploadedImages = new ArrayList<>();
+    UploadImageAdapter adapter;
+    private final String CAMERA = "camera";
+    private final String GALLERY = "gallery";
+    ActivityResultLauncher<Intent> cameraActivityResultLauncher;
+    ActivityResultLauncher<String> galleryActivityResultLauncher;
+    File storageDir;
```
Comment (artur-varosyan) [inline app/src/main/java/com/example/ar_reshare/AddProduct.java:41]: Can you add visibility specifiers to some of these please? e.g. public vs private

**Review by Manoutofpower — APPROVED** [review]

**Review by artur-varosyan — CHANGES_REQUESTED** [review]
Can you please just implement this one change for me? Otherwise looks great!

File: app/src/main/java/com/example/ar_reshare/AddProduct.java (hunk 1 above)
Comment (HellinSun) [reply app/src/main/java/com/example/ar_reshare/AddProduct.java]: it's added now! thanks!

**Review by artur-varosyan — APPROVED** [review]
Thanks, Hellin! It's a yes from me

**Pull request merged**